_posts/2018-10-01-asdf-version-management.md

```diff
@@ -0,0 +1,136 @@
+excerpt: >
+  Take a dive into flexible version management of Elixir, Erlang, and OTP with `asdf`!
+---
+
+# Agnostic Version Management With ASDF
+
+## What is it?
+Oftentimes we need to use multiple versions of our tools.
+Many communities have their own things to do this.
+In Ruby we have `chruby`, `rbenv`, `rvm` and more, NodeJS has `nvm`.
+These tools allow us to easily and quickly switch what we are using for a given project or environment.
+
+Today were going to talk about my favorite version manager of choice, `asdf`, because it lets you manage multiple languages with just one tool because it is agnostic as to what you manage the version of with it.
+There is one big win that I see with `asdf` that no other tool has allowed me to do as easily: control which version of OTP my Elixir was compiled with, and manage that and several versions of Elixir + OTP together.
+Let's check it out!
+
+## Setup
+Installing `asdf` is a breeze.
+
+On macOS:
+
+```shell
+echo -e '\n. $HOME/.asdf/asdf.sh' >> ~/.bash_profile
+echo -e '\n. $HOME/.asdf/completions/asdf.bash' >> ~/.bash_profile
+```
+
+On linux (with a standard bash shell):
+
+```shell
+echo -e '\n. $HOME/.asdf/asdf.sh' >> ~/.bashrc
+echo -e '\n. $HOME/.asdf/completions/asdf.bash' >> ~/.bashrc
+```
+
+With ZSH:
+
+```shell
+echo -e '\n. $HOME/.asdf/asdf.sh' >> ~/.zshrc
+echo -e '\n. $HOME/.asdf/completions/asdf.bash' >> ~/.zshrc
+```
+
+With Fish:
+
+```shell
+echo 'source ~/.asdf/asdf.fish' >> ~/.config/fish/config.fish
+mkdir -p ~/.config/fish/completions; and cp ~/.asdf/completions/asdf.fish ~/.config/fish/completions
+```
+Now restart your shell, and type `asdf` and we get our first introduction to the tool.
+
+```shell
+asdf
+
+MANAGE PLUGINS
+  asdf plugin-add <name> [<git-url>]   Add a plugin from the plugin repo OR, add a Git repo
+                                       as a plugin by specifying the name and repo url
+  asdf plugin-list                     List installed plugins
+  [...]
+
+MANAGE PACKAGES
+  asdf install <name> <version>        Install a specific version of a package or,
+                                       with no arguments, install all the package
+                                       versions listed in the .tool-versions file
+  asdf uninstall <name> <version>      Remove a specific version of a package
+  asdf current                         Display current version set or being used for all packages
+  asdf current <name>                  Display current version set or being used for package
+  [...]
+
+UTILS
+  asdf reshim <name> <version>         Recreate shims for version of a package
+  asdf update                          Update asdf to the latest stable release
+  asdf update --head                   Update asdf to the latest on the master branch
+```
+
+## Using it with Elixir
+To get `asdf` working with Elixir, we first will need Erlang.
+Depending on our system, there are some simple steps:
+
+On OSX:
+
+```shell
+brew install autoconf
```

Comment:
Most people would probably also like to have Wx installed so they can run the Observer GUI. `brew install wxmac`

Reply:
Awesome, will add.